### ISSUE TITLE
"possible EventEmitter memory leak detected" even using promises

### ISSUE THREAD

**Issue by kblok** [norberteder/trello#5]
I am getting this "possible EventEmitter memory leak detected" when I try to create more than 10 cards at the same time.
This could be the expected behavior but the thing is that I am using promises so I am chaining the card creation in this way:

```
var promise;
var cardCreationPromise;
_.each(myData, function (ticket) {
                if(!promise) {
                    cardCreationPromise = Q.defer();
                    trello.addCard(ticket.summary, ticket.description, listId, cardCreationPromise.makeNodeResolver());
                    promise = cardCreationPromise.promise;
                } else {
                    promise = promise.then(function(e) {
                        cardCreationPromise = Q.defer();
                        trello.addCard(ticket.summary, ticket.description, listId, cardCreationPromise.makeNodeResolver());
                        return cardCreationPromise.promise;
                    });    
                }
            });
```

Typical chaining...

After some research I have learned that this "memory leak issue "is a common problem in Restler and in  Node JS in general.
You can do your own research and see if you end up with the same solution as I. The main problem is that when we use the "on" event, the listener is still registered so we are "leaking" (It is not technically a leak) listeners even if we chain the calls. But if you use "once" instead of "on" we are calling to the next card creating after the listener is cleared, and we should clear the "error" listener too:

```
function makeRequest(fn, uri, options, callback) {
    var request = fn(uri, options);
    request.once('complete', function (result) {
        if (result instanceof Error) {
            callback(result);
        } else {
            callback(null, result);
        }
        request.removeAllListeners('error');
    });
}
```

These two changes fix the issue in my scenario.

**Comment by tgirardi** [comment]
+1 similar problem: when I try to make a request after another request, the "on complete" listener from the first one is still active, so the callback for the second request receives the result from the first request.

**Comment by sinisterstuf** [comment]
This is fixed by #9 and #13.

**Comment by sinisterstuf** [comment]
The tests to confirm it are currently marked as pending because they require work to reproduce, but I'm working on generalising them.
